### PR TITLE
Added Azure.AKS.PodIdentity rule #991

### LINKED ISSUE #991: Use Azure AD Pod Managed Identities (preview)
# Rule request

## Suggested rule change

AKS clusters should enable and use AAD pod identities. AAD pod identity provide a native way to use managed identities for workloads hosted on AKS.

This is a preview feature.

## Applies to the following

The rule applies to the following:

- Resource type: **Microsoft.ContainerService/managedClusters**

## Additional context

[Identity-based authentication](https://docs.microsoft.com/azure/architecture/framework/security/design-identity-authentication#use-identity-based-authentication)

### PR DESCRIPTION
**Pull request by ArmaanMcleod**
## PR Summary

Fix #991

Added `Azure.AKS.PodIdentity` rule. 

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

File: src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml [hunk 1]
```diff
@@ -97,6 +97,22 @@ spec:
...
+# Synopsis: Configure AKS clusters to use AAD pod identities to access Azure resources securely.
+apiVersion: github.com/microsoft/PSRule/v1
+kind: Rule
+metadata:
+  name: Azure.AKS.PodIdentity
+  tags:
+    release: 'preview'
+    ruleSet: '2021_12'
+spec:
+  type:
```
Comment (BernieWhite) [inline src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml:110]: @ArmaanMcleod Lets add a selector to only suggest this rule if Azure CNI is used. The latest guidance does not recommend this for kubenet clusters.

https://docs.microsoft.com/azure/aks/use-azure-ad-pod-identity#using-kubenet-network-plugin-with-azure-active-directory-pod-managed-identities

Selector example:

https://github.com/Azure/PSRule.Rules.Azure/blob/816c5c5086cd1c300f02781f9e8a952c9011bc10/src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml#L44-L60

https://github.com/Azure/PSRule.Rules.Azure/blob/816c5c5086cd1c300f02781f9e8a952c9011bc10/src/PSRule.Rules.Azure/rules/Azure.AKS.Rule.yaml#L104-L116

File: docs/en/rules/Azure.AKS.PodIdentity.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+severity: Important
+pillar: Operational Excellence
+category: Configuration
```
Comment (BernieWhite) [inline docs/en/rules/Azure.AKS.PodIdentity.md:4]:
```suggestion
pillar: Security
category: Authentication
```

File: docs/en/rules/Azure.AKS.PodIdentity.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Administrators create identities and bindings as Kubernetes primitives that allow pods to access
+Azure resources that rely on Azure AD as an identity provider.
+
+## RECOMMENDATION
+
+Consider enabling AAD pod identities on AKS clusters.
+
+## EXAMPLES
+
```
Comment (BernieWhite) [inline docs/en/rules/Azure.AKS.PodIdentity.md:27]: Would be good to have a Bicep and Template example. Maybe just update the docs code and build to get both.

https://github.com/Azure/PSRule.Rules.Azure/blob/816c5c5086cd1c300f02781f9e8a952c9011bc10/docs/examples-aks.bicep

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod Nice work! A few suggestions but otherwise good to go.

**Comment by ArmaanMcleod** [comment]
> @ArmaanMcleod Nice work! A few suggestions but otherwise good to go.

Thanks @BernieWhite. Made some changes in regards to your feedback. Let me know what you think.

**Review by BernieWhite — APPROVED** [review]
@ArmaanMcleod All good to merge!

**Pull request merged**